docs/testplan/Smartswitch-test-plan.md

```diff
@@ -645,6 +672,672 @@ root@sonic:/home/cisco# show chassis modules status
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco# nohup bash -c "sleep 5 && tail /dev/zero" &
+root@sonic:/home/cisco# nohup: ignoring input and appending output to 'nohup.out'
+root@sonic:/home/cisco#
+.
+(Going for reboot)
+.
+root@sonic:/home/cisco# show chassis modules status
+  Name           Description    Physical-Slot    Oper-Status    Admin-Status           Serial
+------  --------------------  ---------------  -------------  --------------  ---------------
+  DPU0  Data Processing Unit              N/A        Offline            down  154226463179136
+  DPU1  Data Processing Unit              N/A        Offline            down  154226463179152
+  DPU2  Data Processing Unit              N/A        Offline            down  154226463179168
+  DPUX  Data Processing Unit              N/A        Offline            down  154226463179184
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco# config chassis startup DPU0
+root@sonic:/home/cisco# config chassis startup DPU1
+root@sonic:/home/cisco# config chassis startup DPU2
+root@sonic:/home/cisco# config chassis startup DPUX
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco#
+root@sonic:/home/cisco# show chassis modules status
+  Name           Description    Physical-Slot    Oper-Status    Admin-Status           Serial
+------  --------------------  ---------------  -------------  --------------  ---------------
+  DPU0  Data Processing Unit              N/A         Online              up  154226463179136
+  DPU1  Data Processing Unit              N/A         Online              up  154226463179152
+  DPU2  Data Processing Unit              N/A         Online              up  154226463179168
+  DPUX  Data Processing Unit              N/A         Online              up  154226463179184
+
+root@sonic:/home/cisco# ping 169.254.200.1
+PING 169.254.200.1 (169.254.200.1) 56(84) bytes of data.
+64 bytes from 169.254.200.1: icmp_seq=1 ttl=64 time=0.160 ms
+^C
+--- 169.254.28.1 ping statistics ---
+1 packets transmitted, 1 received, 0% packet loss, time 0ms
+rtt min/avg/max/mdev = 0.160/0.160/0.160/0.000 ms
+root@sonic:/home/cisco# 
+```
+ 
+#### Pass/Fail Criteria
+
+ * Verify number of DPUs from inventory file for the testbed and number of DPUs shown in the cli output.
+ * Verify Ping works to all the mid plane ip listed in the ansible inventory file for the testbed.
+
+
+### 1.15 Check DPU status and pcie Link after memory exhaustion on DPU
+
+#### Steps
+
+- DPU:
+   * Use `sudo swapoff -a`. Swapping is turned off so the OOM is triggered in a shorter time.
+   * Use 'nohup bash -c "sleep 5 && tail /dev/zero" &' to to run out of memory completely.
+   * It runs on the background and `nohup` is also necessary to protect thebackground process.
+   * Added `sleep 5` to ensure ansible receive the result first.
+   
+    - Switch:
+       * Powercycling of DPU is to ensure that pcie link came up properly after the memory exhaustion test.
```

Comment:
In case of test failure for all these PCIe Link check, can we make sure of add some diagnostics logs for easy troubleshooting? Either you can mention here or can be added when writing test case.

Reply:
Yes, I will make sure to have logs included in test scripts.